beetsplug/audible.py

```diff
@@ -104,11 +248,117 @@ def __init__(self):
+        # Warning, this method is rather messy, and it's easy for this to go wrong.
+        # This should be used as a last resort
+        affixes = find_regular_affixes([c.title for c in items])
+
+        all_remote_chapters: List = deepcopy(album.tracks)
+        matches = []
+        for chapter in items:
+            # TODO: need a string distance algorithm that penalises number replacements more
+            best_matches = list(
+                sorted(
+                    all_remote_chapters,
+                    key=lambda c: specialised_levenshtein(chapter.title, c.title, affixes),
+                )
+            )
+            try:
+                best_match = best_matches[0]
+            except IndexError:
+                return None
+            matches.append(best_match)
+            all_remote_chapters.remove(best_match)
+        return matches
+
+    def attempt_match_single_item(self, items: List[Item], album: AlbumInfo) -> Optional[List[Item]]:
+        """If the input album has a single item, use that; if the album also has a single item, prefer that."""
+        if len(items) == 1:
+            # Prefer a single named book from the remote source
+            if len(album.tracks) == 1 and album.tracks[0].title != "Chapter 1":
+                matches = album.tracks
+            else:
+                matches = items
+            return matches
+
+    def sort_tracks(self, album: AlbumInfo, items: List[Item]) -> Optional[List[TrackInfo]]:
+        common_attrs = get_common_data_attributes(album.tracks[0])
+
+        # this is the master list of different approaches
+        # must be updated for any additional options added in the future
+        possible_matching_algorithms = {
+            "single_file": self.attempt_match_single_item,
+            "source_numbering": self.attempt_match_trust_source_numbering,
+            "starting_numbers": self.attempt_match_starting_numbers,
+            "natural_sort": self.attempt_match_natsort,
+            "chapter_levenshtein": self.attempt_match_chapter_levenshtein,
+        }
+        for algorithm_choice in self.config["chapter_matching_algorithms"]:
+            algorithm_choice = str(algorithm_choice)
+            if algorithm_choice not in possible_matching_algorithms:
+                self._log.error(
+                    f"'{algorithm_choice}' is not a valid algorithm choice for chapter matching; there are {len(possible_matching_algorithms.keys())}"
+                )
+                continue
+            function = possible_matching_algorithms[algorithm_choice]
+            matches = function(items, album)
+            if matches is not None:
+                tracks = convert_items_to_trackinfo(matches, common_attrs)
+                return tracks
+        # if len(items) > len(album.tracks):
+        #     # TODO: find a better way to handle this
+        #     # right now just reject this match
+        #     return None
```

Comment:
Since this is commented out, what happens in this situation? `None` should still be returned but the item would fail to match.

I believe this case is currently being handled correctly in the existing matching algorithm.